clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -1045,7 +1045,9 @@ letCast _ e = return e
 -- and expression where two casts are "back-to-back" after which we can
 -- eliminate them in 'eliminateCastCast'.
 argCastSpec :: HasCallStack => NormRewrite
-argCastSpec ctx e@(App _ (stripTicks -> Cast e' _ _)) = do
+argCastSpec ctx e@(App f (stripTicks -> Cast e' _ _))
+ -- Don't specialise prims, because we can't push casts into them
+ | not . isPrim . fst . collectArgs $ f = do
```

Comment:
Since we're fixing this function anyway, another thing we want to avoid is specializing when `e'` is itself a `Cast`. `eliminateCastCast` will eliminate that cast-of-cast, but if we stop `argCastSpec` from matching on it we can do the `eliminateCastCast` in the "current" function instead of the specialized function.

Reply:
Would you mind making a patch?

Reply:
I'll make a patch